R/dist_quantiles.R

```diff
@@ -5,6 +5,9 @@ new_quantiles <- function(values = double(), quantile_levels = double()) {
   vec_cast(values, double())
   vec_cast(quantile_levels, double())
```

Comment:
(this is a no-op. `vec_cast` doesn't have side effects)

Reply:
I'm not positive this is the "best" thing to do. It doesn't need to convert. I'm open to suggestions.

Reply:
I'm not sure.  It certainly seems simpler to think about if we could go ahead and cast.  But I'm not sure if there's a situation where someone really really wants their integer-type/esoteric-yet-still-double-compatible quantiles.